sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsublite/ReadWriteIT.java

```diff
@@ -236,6 +236,8 @@ public void testReadWrite() throws Exception {
     pipeline.getOptions().as(TestPipelineOptions.class).setBlockOnRun(false);
 
     TopicPath topic = createTopic(getProject(pipeline.getOptions()));
+    // Sleep for topic creation to propagate.
+    Thread.sleep(10000);
```

Comment:
Generally, its better to not add sleep to code like this, but rather some logic that checks if a condition is met.

What if the sleep is not long enough in all situations?

This seems like it would introduce flakey/incorrect behavior

Reply:
I've made this into a loop which will try up to 30 times, retrying errors after a 1 second sleep. It builds in a longer potential latency without impacting the average case.

Speaking now with system knowledge, the read which is flaky uses bounded staleness of 30 seconds, which means this loop should always outlive the race condition.

Reply:
To be more specific: there's no check that can be made conclusively that the topic will be found when you go to create the subscription. Just because the server that you make a GetTopic request for example can get the topic doesn't mean a subsequent subscription creation request is guaranteed to succeed.